rfq/negotiator.go

```diff
@@ -769,9 +791,31 @@ func (n *Negotiator) HandleIncomingSellAccept(msg rfqmsg.SellAccept,
 		tolerance := rfqmath.NewBigIntFromUint64(
 			n.cfg.AcceptPriceDeviationPpm,
 		)
-		acceptablePrice := msg.AssetRate.Rate.WithinTolerance(
+		acceptablePrice, err := msg.AssetRate.Rate.WithinTolerance(
 			assetRate.Rate, tolerance,
 		)
+		if err != nil {
+			// The tolerance check failed. We will return without
+			// calling the quote accept callback.
+			err = fmt.Errorf("failed to check tolerance: %w", err)
+			log.Errorf("Error checking tolerance: %v", err)
+			n.cfg.ErrChan <- err
```

Comment:
I'm not sure we should shut down here (which happens when we send an error on the main channel). Otherwise a "rogue" oracle can cause your node to go down.